src/main/java/seedu/address/logic/commands/SortCommand.java

```diff
@@ -0,0 +1,191 @@
+            + ": Sorts all restaurants in ascending or descending order of its aggregate ratings.\n"
+            + "The list of restaurants will be sorted in descending order of aggregate ratings by default, "
+            + "unless its order (ASC/DES) is specified. You can limit the number of ratings by specifying the limit.\n"
+            + "The specified keywords for order are 'ASC' or 'DES' (case-insensitive) and represent the order "
+            + "that the restaurants will be listed in.\n"
+            + "Parameters: [or/ASC] or [or/DES] and [l/LIMIT]\n"
+            + "Example: " + COMMAND_WORD + " " + PREFIX_ORDER + "asc " + PREFIX_LIMIT + "3";
+
+    public static final String MESSAGE_SUCCESS_ALL = "Sorted all restaurants in %1$s order";
+    public static final String MESSAGE_SUCCESS_LIMIT = "Sorted %1$s restaurant(s) in %2$s order";
+
+    private Order order;
+    private Optional<Limit> limit;
+    private String commandMessage;
+
+    /**
+     * Constructs a SortCommand with optional limit.
+     */
+    public SortCommand(Order order, Optional<Limit> limit) {
+        this.order = order;
+        if (limit.isPresent()) {
+            this.limit = Optional.ofNullable(limit.get());
+            if (order.equals(new Order(ORDER_ASC))) {
+                this.commandMessage = String.format(MESSAGE_SUCCESS_LIMIT,
+                        "the top " + limit.get().toInteger(), FEEDBACK_ASC);
+            } else {
+                this.commandMessage = String.format(MESSAGE_SUCCESS_LIMIT,
+                        "the top " + limit.get().toInteger(), FEEDBACK_DES);
+            }
+        } else {
+            if (order.equals(new Order(ORDER_ASC))) {
+                this.commandMessage = String.format(MESSAGE_SUCCESS_ALL, FEEDBACK_ASC);
+            } else {
+                this.commandMessage = String.format(MESSAGE_SUCCESS_ALL, FEEDBACK_DES);
+            }
+        }
+    }
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) {
+
+        requireNonNull(model);
+        model.sortRestaurantList(new SortRating(order));
+
+        // Filter the sorted list to only show those within the limited range
+        if (limit == null) {
+            model.updateFilteredRestaurantList(PREDICATE_SHOW_ALL_RESTAURANTS);
+        } else {
+            int limitInt = limit.get().toInteger();
+            ArrayList<Float> uniqueRatings = model.getUniqueRatings();
+
+            // If limit < number of Restaurants, need to filter
+            if (limitInt < uniqueRatings.size()) {
+                float ratingBorder = model.getUniqueRatings().get(limitInt);
+
+                if (order.equals(new Order(ORDER_ASC))) {
+                    Predicate<Restaurant> predicateShowLimitedAsc = (r) ->
+                            r.getSummary().getAvgRating() < ratingBorder;
+                    model.updateFilteredRestaurantList(predicateShowLimitedAsc);
+                } else {
```

Comment:
there's a few nested if else here, is there a way to change the logic flow so that it is better structured? 
https://nus-cs2103-ay1819s2.github.io/cs2103-website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting

Reply:
Yes you're right! I'll change modify them in my next PR. Thanks for the tip!